frinx/common/worker/worker.py

```diff
@@ -102,22 +109,39 @@ def execute(self, task: Task) -> TaskResult:
 
     @classmethod
     def _execute_wrapper(cls, task: RawTaskIO) -> Any:
+
+        task_type = task.get('taskType')
+        increment_task_poll(metrics, task_type)
+
         try:
             cls.WorkerInput.parse_obj(task['inputData'])
         except ValidationError as error:
             logger.error('Validation error occurred: %s', error)
             return TaskResult(status=TaskResultStatus.FAILED, logs=[TaskExecLog(str(error))]).dict()
+
         try:
-            # TODO check if ok
             logger.debug('Executing task %s:', task)
-            task_result = cls.execute(cls, Task(**task)).dict()  # type: ignore[arg-type]
+            task_result: TaskResult = cls._execute_func(task)
             logger.debug('Task result %s:', task_result)
             return task_result
 
         except Exception as error:
+            increment_task_execution_error(metrics, task_type, error)
+            increment_uncaught_exception(metrics)
             logger.error('Validation error occurred: %s', error)
             return TaskResult(status=TaskResultStatus.FAILED, logs=[TaskExecLog(str(error))]).dict()
 
+    @classmethod
+    def _execute_func(cls, task: RawTaskIO) -> TaskResult:
+        if not metrics.settings.metrics_enabled:
+            return cls.execute(cls, Task(**task)).dict()  # type: ignore[arg-type]
+
+        start_time = time.time()
+        task_result: TaskResult = cls.execute(cls, Task(**task)).dict()  # type: ignore[arg-type]
+        finish_time = time.time()
+        record_task_execute_time(metrics, task.get('taskType'), finish_time - start_time)
+        return task_result
+
```

Comment:
I was thinking again about this. Maybe a set of decorators would fit here better. But its just an idea.